drivers/media/i2c/imx477.c

```diff
@@ -2090,6 +2138,12 @@ static int imx477_probe(struct i2c_client *client)
 
 	v4l2_i2c_subdev_init(&imx477->sd, client, &imx477_subdev_ops);
 
+	match = of_match_device(imx477_dt_ids, dev);
+	if (!match)
+		return -ENODEV;
+	memcpy(&imx477->compatible_data, match->data,
+	       sizeof(imx477->compatible_data));
```

Comment:
Why memcpy instead of having ```struct imx477_compatible_data *compatible_data;``` in the struct and copy the pointer?

Reply:
True enough, will fix that!